File: Solutions/elixir/all-your-base/lib/all_your_base.ex
defmodule AllYourBase do
  @doc """
  Given a number in input base, represented as a sequence of digits, converts it to output base,
  or returns an error tuple if either of the bases are less than 2
  """

  @spec convert(list, integer, integer) :: {:ok, list} | {:error, String.t()}
  def convert(_, input_base, _) when input_base <2, do: {:error, "input base must be >= 2"}
  def convert(_, _, output_base) when output_base <2, do: {:error, "output base must be >= 2"}
  def convert(digits, input_base, output_base) do
    if valid_digits?(digits, input_base) do
      {:error, "all digits must be >= 0 and < input base"}
    else
      result = digits
      |> from_base(input_base)
      |> to_base(output_base, [])
      {:ok, result}
    end
  end
  defp valid_digits?(digits, input_base) do
    Enum.any?(digits, &(&1 < 0 or &1 >= input_base))
  end
  defp from_base(digits, input_base) do
    Enum.reduce(digits, 0, fn digit, acc -> acc * input_base + digit end)
  end
  defp to_base(0, _, []), do: [0]
  defp to_base(0, _, acc), do: acc
  defp to_base(number, output_base, acc) do
    div = div(number, output_base)
    rem = rem(number, output_base)
    to_base(div, output_base, [rem | acc])
  end
end

File: Solutions/python/all-your-base/all_your_base.py
def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError("input base must be >= 2")
    if output_base < 2:
        raise ValueError("output base must be >= 2")
    if any(d < 0 or d >= input_base for d in digits):
        raise ValueError("all digits must satisfy 0 <= d < input base")
    # Convert input digits to base 10
    num = sum(d * input_base**i for i, d in enumerate(reversed(digits)))
    if num == 0:
        return [0]

    # Convert base 10 number to output base
    output_digits = []
    while num > 0:
        num, remainder = divmod(num, output_base)
        output_digits.append(remainder)

    return output_digits[::-1]
